### PR TITLE
GPS Stuff

### PR DESCRIPTION
**Pull request by Geevies**
* GPSs now have an idle mode, where they will update their transmitted location every six seconds. This mode activates if they're not in someone's possession.
* You can now keep your GPS in your bag, webbing, etc. It will keep actively updating as you move. If you drop the bag, webbing, etc., it'll swap into idle mode after you move again.
* Ported the GPS compass, selecting a GPS to track on the C-Track column will add it to the compass. The GPS must be in your active hand/slot to show you the compass.

Compass Ported from: https://github.com/PolarisSS13/Polaris/pull/7928

Port disclaimer: Just about everything in the Compass modules folder is ported straight from Polaris, but most of the bits in gps.dm was mangled into place by myself. If any of those parts are broken, it's cuz of me.

### Pictures!

image
image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Review by JohnWildkins — APPROVED** [review]
Seems to work good, compass is cool.
I feel like the track/untrack all system is a little unintuitive, but it works well enough. The only other thing is that maybe GPSes with the same turf shouldn't appear on the compass? Right now they are locked dead south which might be confusing.

File: code/modules/telesci/gps.dm [hunk 1]
```diff
@@ -10,19 +10,28 @@ var/list/GPS_list = list()
...
 	var/gpstag = "COM0"
-	var/emped = 0
-	var/held_by = null
-	var/implanted_into = null
+	var/compass_color = "#193A7A"
+
+	var/emped = FALSE
+
+	var/mob/held_by = null
+	var/mob/implanted_into = null
```
Comment (NonQueueingMatt) [inline code/modules/telesci/gps.dm:18]: need to clean up both of these in Destroy

**Review by NonQueueingMatt — DISMISSED** [review]

File: code/modules/telesci/gps.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/telesci/gps.dm:18]: already handled
image

**Review by Casper3667 — APPROVED** [review]

**Pull request merged**